src/capi/azext_capi/custom.py

```diff
@@ -469,8 +468,8 @@ def create_workload_cluster(  # pylint: disable=too-many-arguments,too-many-loca
         defined_args = [v["name"] for v in mutual_exclusive_args if v["value"]]
         if defined_args:
             defined_args = " ,".join(defined_args)
-            error_msg = f'The following arguments are incompatible with "--template":\n{defined_args}'
-            raise MutuallyExclusiveArgumentError(error_msg)
+            warning_msg = f'The following arguments are incompatible with "--template":\n{defined_args}'
+            logger.warning(warning_msg)
```

Comment:
Why is this change needed? Did the function use to terminate after the error?

Reply:
Yes, sorry this is a logic change that I should have explained up front. This used to be a hard error, but in testing the external-cloud-provider changes in the pipeline I realized we use a local template there but now want to set the `--external-cloud-provider` flag. 

So I decided to make all these cases a yellow warning message rather than fail. It's basically saying if your user-supplied template doesn't have sections like 

```jinja2
{% if EXTERNAL_CLOUD_PROVIDER %}external{% else %}azure{% endif %}
```

then it's not likely to work with those flags.